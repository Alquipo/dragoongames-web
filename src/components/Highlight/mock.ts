export default {
  title: 'Read Dead is back!',
  subtitle: "come see John's new adventures",
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'buy now',
  buttonLink: '/games/rdr2'
}
