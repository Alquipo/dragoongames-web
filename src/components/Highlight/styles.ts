import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { HighlightProps } from '.'

type WrapperProps = Pick<HighlightProps, 'backgroundImage'>

export const Wrapper = styled.section<WrapperProps>`
  ${({ backgroundImage }) => css`
    position: relative;

    background-image: url(${backgroundImage});
    background-position: center center;
    background-size: cover;

    height: 23rem;
    display: grid;

    &::after {
      content: '';
      position: absolute;
      width: 100%;
      height: 100%;
      background-color: rgba(0, 0, 0, 0.6);
    }

    ${media.greaterThan('medium')`
      height: 32rem;
    `}
  `}
`
export const Content = styled.div`
  ${({ theme }) => css`
    z-index: ${theme.layers.base};
    text-align: right;
    padding: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      align-self: end;
      padding: ${theme.spacings.large};
    `}
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    color: ${theme.colors.white};
    margin-bottom: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`
