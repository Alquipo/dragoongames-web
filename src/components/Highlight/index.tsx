import * as S from './styles'

const Highlight = () => (
  <S.Wrapper>
    <h1>Highlight</h1>
  </S.Wrapper>
)

export default Highlight
