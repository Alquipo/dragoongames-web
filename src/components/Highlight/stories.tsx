import { Story, Meta } from '@storybook/react/types-6-0'
import Highlight from '.'

export default {
  title: 'Highlight',
  component: Highlight
} as Meta

export const Default: Story = () => <Highlight />
