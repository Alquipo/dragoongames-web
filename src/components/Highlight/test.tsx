import { render, screen } from '@testing-library/react'

import Highlight from '.'

describe('<Highlight />', () => {
  it('should render the heading', () => {
    const { container } = render(<Highlight />)

    expect(
      screen.getByRole('heading', { name: /Highlight/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
