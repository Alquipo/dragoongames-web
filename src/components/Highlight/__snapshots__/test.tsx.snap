// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Highlight
  </h1>
</main>
`
